src/c++/library/common.h

```diff
@@ -330,10 +330,17 @@ class InferInput {
   /// \return Error object indicating success or failure.
   Error ByteSize(size_t* byte_size) const;
 
+  /// \return true if this input should be sent in binary format.
+  bool BinaryData() const { return binary_data_; }
+
+  /// \return Error object indicating success or failure.
+  Error SetBinaryData(const bool binary_data);
```

Comment:
Sure, this could return void. I was just trying to follow the pattern. I'm not a fan of Error returns anyway.

Reply:
This is already anti-pattern, ideally this would be set in the constructor but that seemed messier.

Reply:
What do you want me to do? Void or leave as is?

Reply:
Void. If not, you would need to check the status after it is used every time.

Reply:
Changed to void 👍

Reply:
@Tabrizian , @tgerdesnv Any concern with keeping consistency with other function definitions here? I see in general we have the pattern of passing back an Error for all set methods - even if there isn't a specific error condition. I agree that its not needed in many cases - but the pattern is established. Maybe we would do error for now and file a ticket to change the pattern? Or otherwise - change to void and another ticket to change in other places. Whenever I'm reading code and I see that "one of these is different" - I start to wonder why - in this case I don't see it as different than the other setters / getters (maybe I'm missing something) - not a blocker

Reply:
If we return Error status then
#1 - we need to test it
#2 - we need to consume and use it on every function call that returns a status

I suspect there are many cases that should be changed from Error status to void, and many more valid cases that need to be tested and actually consumed.  Plenty of follow-up work to do here. @debermudez please file follow-up tickets for cleaning up the return values of clients.

Reply:
@Tabrizian Do we have a style guide / doc that covers this? If we are concerned about removing it in other spots - I would vote for leaving it as Error for now - and then maybe updating the API and having a versioning scheme around it similar to C API -

Reply:
We don't have a guide that covers this. Yes, I think that would be the best course of action for now.

Reply:
What's the final recommendation on this? Return Error or void?

Reply:
> What's the final recommendation on this? Return Error or void?

@tgerdesnv  Would you be ok with returning error for now until we can change the rest of the apis?

Reply:
Yes that is fine. We still need a ticket for testing though.

Reply:
Changed to error 👍